Dart/thirdPartySrc/analysisServer/com/google/dart/server/internal/remote/RemoteAnalysisServerImpl.java

```diff
@@ -732,6 +814,12 @@ private void processResponse(JsonObject response) throws Exception {
     synchronized (consumerMapLock) {
       consumer = consumerMap.get(idString);
     }
+
+    // handle requests by calling processRequests, return if this was a request input type
+    if(consumer == null && processRequests(response, idString)) {
```

Comment:
Checking consumer for null doesn't seem to be correct. Consumer is found in the map by "id". It's a map of requests that the client has sent to the server.

Spec says: "The ids used by the server are independent of the ids used by the client. In other words, the ids sent by the server might be equal to ids already sent from the client, but are unrelated to the client request with that same id."

I removed `consumer == null` and moved processRequests() 3 lines up, right after `String idString = idJsonPrimitive.getAsString()`